components/tools/OmeroPy/test/integration/clitest/test_duplicate.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+# Copyright (C) 2016 University of Dundee & Open Microscopy Environment.
+# All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import omero
+from omero.plugins.duplicate import DuplicateControl
+from test.integration.clitest.cli import CLITest
+import pytest
+
+object_types = ["Image", "Dataset", "Project", "Plate", "Screen"]
+model = ["", "I"]
+
+
+class TestDuplicate(CLITest):
+
+    def setup_method(self, method):
+        super(TestDuplicate, self).setup_method(method)
+        self.cli.register("duplicate", DuplicateControl, "TEST")
+        self.args += ["duplicate"]
+
+    def duplicate(self, capfd):
+        self.cli.invoke(self.args, strict=True)
+        return capfd.readouterr()[0]
+
+    def get_dataset(self, iid):
+        """Retrieve all the parent datasets linked to the image"""
+
+        params = omero.sys.ParametersI()
+        params.addId(iid)
+        query = ("select d from Dataset as d where exists"
+                 "(select l from DatasetImageLink as l where "
+                 "l.child.id=:id and l.parent=d.id)")
+        return self.query.findAllByQuery(query, params)
+
+    def get_project(self, did):
+        """Retrieve all the parent projects linked to the dataaset"""
+
+        params = omero.sys.ParametersI()
+        params.addId(did)
+        query = ("select p from Project as p where exists"
+                 "(select l from ProjectDatasetLink as l where "
+                 "l.child.id=:id and l.parent=p.id)")
```

Comment:
this might simply be something like,
`SELECT DISTINCT parent FROM ProjectDatasetLink WHERE child.id = :id`